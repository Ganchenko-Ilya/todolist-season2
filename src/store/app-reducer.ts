import { PayloadAction, createSlice } from "@reduxjs/toolkit";

export type StatusType = "idle" | "loading" | "succeeded" | "failed";

export type messageInfo = {
  errorInfo?: null | string;
  succeededInfo?: null | string;
};
export type InitialStateType = {
  errorInfo: null | string;
  succeededInfo: null | string;
  status: StatusType;
};
const initialState: InitialStateType = {
  errorInfo: null,
  succeededInfo: null,
  status: "idle",
};

const slice = createSlice({
  initialState,
  name: "app",
  reducers: {
    setInfoAC: (state, action: PayloadAction<{ message: messageInfo }>) => {
      return { ...state, ...action.payload.message };
    },
    changeAppStatusAC: (state, action: PayloadAction<{ status: StatusType }>) => {
      state.status = action.payload.status;
    },
  },
});

export const appReducer = slice.reducer;
export const { changeAppStatusAC, setInfoAC } = slice.actions;
export type ActionsAppType = ReturnType<(typeof slice.actions)[keyof typeof slice.actions]>;

// export type ActionsAppType = serErrorTypeAC | changeAppStatusTypeAC;

// export const appReducer = (state: InitialStateType = initialState, action: ActionsAppType) => {
//   switch (action.type) {
//     case "SET-INFO": {
//       return { ...state, ...action.message };
//     }
//     case "CHANGE-STATUS-APP": {
//       return { ...state, status: action.status };
//     }
//     default:
//       return state;
//   }
// };

// export const setInfoAC = (message: messageInfo) => ({ type: "SET-INFO", message }) as const;
// export const changeAppStatusAC = (status: StatusType) => ({ type: "CHANGE-STATUS-APP", status }) as const;

// type serErrorTypeAC = ReturnType<typeof setInfoAC>;
// type changeAppStatusTypeAC = ReturnType<typeof changeAppStatusAC>;
