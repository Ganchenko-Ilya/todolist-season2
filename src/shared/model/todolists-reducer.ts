import { PayloadAction, createSlice } from "@reduxjs/toolkit";
import { todolistsApi } from "../api/todolistsApi";
import { ChangeTodolistsResponseType, StatusType, TodolistsResponse } from "shared/types";
import { createAppAsyncThunk, errorCatchHelpFunc, helpChangeStatusFunc, helpResultFunc } from "shared/utils";
import { changeAppStatusAC, setInfoAC } from "./app-reducer";
import { authThunk } from "./auth-reducer";

export const initialStateTodo: ChangeTodolistsResponseType[] = [];

export type ActionsTodolistsType = ReturnType<(typeof slice.actions)[keyof typeof slice.actions]>;

const slice = createSlice({
  initialState: initialStateTodo,
  name: "todolists",
  reducers: {
    changeStatusTodoAC: (state, action: PayloadAction<{ tId: string; status: StatusType }>) => {
      const index = state.findIndex((el) => el.id === action.payload.tId);
      if (index > -1) {
        state[index].statusTodo = action.payload.status;
      }
    },
  },
  extraReducers: (builder) => {
    builder
      .addCase(authThunk.loginOutTC.fulfilled, () => {
        return [];
      })
      .addCase(setTodo.fulfilled, (_, action) => {
        return action.payload.todolists.map((el) => ({ ...el, statusTodo: "idle" }));
      })
      .addCase(deleteTodo.fulfilled, (state, action) => {
        const index = state.findIndex((el) => el.id === action.payload.tId);
        if (index > -1) {
          state.splice(index, 1);
        }
      })
      .addCase(addTodo.fulfilled, (state, action) => {
        state.unshift({
          id: action.payload.tId,
          title: action.payload.title,
          addedDate: "",
          order: 0,
          statusTodo: "idle",
        });
      })
      .addCase(editTitleTodo.fulfilled, (state, action) => {
        const index = state.findIndex((el) => el.id === action.payload.tId);
        if (index > -1) {
          state[index].title = action.payload.title;
        }
      });
  },
});

export const { changeStatusTodoAC } = slice.actions;
export const todolistReducer = slice.reducer;

const setTodo = createAppAsyncThunk<{ todolists: TodolistsResponse[] }>(
  `${slice.name}/setTodo`,
  async (_, thunkApi) => {
    const { dispatch, rejectWithValue } = thunkApi;
    dispatch(changeAppStatusAC({ status: "loading" }));
    try {
      const res = await todolistsApi.getTodolists();
      dispatch(changeAppStatusAC({ status: "succeeded" }));
      return { todolists: res.data };
    } catch (error) {
      errorCatchHelpFunc(dispatch, error);
      return rejectWithValue(null);
    }
  },
);

const deleteTodo = createAppAsyncThunk<{ tId: string }, { id: string }>(
  `${slice.name}/deleteTodolist`,
  async (param, thunkApi) => {
    const { dispatch, rejectWithValue } = thunkApi;
    helpChangeStatusFunc(param.id, "loading", dispatch);
    try {
      await todolistsApi.deleteTodolist(param.id);
      helpChangeStatusFunc(param.id, "succeeded", dispatch);
      dispatch(setInfoAC({ message: { succeededInfo: "Delete Todo!" } }));
      return { tId: param.id };
    } catch (error) {
      errorCatchHelpFunc(dispatch, error);
      dispatch(changeStatusTodoAC({ tId: param.id, status: "failed" }));
      return rejectWithValue(null);
    }
  },
);

const addTodo = createAppAsyncThunk<{ tId: string; title: string }, { title: string }>(
  `${slice.name}/addTodo`,
  async (param, thunkApi) => {
    const { dispatch, rejectWithValue } = thunkApi;
    dispatch(changeAppStatusAC({ status: "loading" }));
    try {
      const res = await todolistsApi.addTodolist(param.title);
      const resultCode = helpResultFunc<{ item: TodolistsResponse }>(() => {}, res, dispatch, "Add Todo!");
      return resultCode ? rejectWithValue(null) : { tId: res.data.data.item.id, title: param.title };
    } catch (error) {
      errorCatchHelpFunc(dispatch, error);
      return rejectWithValue(null);
    }
  },
);

const editTitleTodo = createAppAsyncThunk<{ tId: string; title: string }, { id: string; title: string }>(
  `${slice.name}/editTitleTodo`,
  async (param, thunkApi) => {
    const { dispatch, rejectWithValue } = thunkApi;
    dispatch(changeAppStatusAC({ status: "loading" }));
    try {
      const res = await todolistsApi.changeTitleTodolist(param.id, param.title);
      const resultCode = helpResultFunc(() => {}, res, dispatch, "Todo changed!");
      return resultCode ? rejectWithValue(null) : { tId: param.id, title: param.title };
    } catch (error) {
      errorCatchHelpFunc(dispatch, error);
      return rejectWithValue(null);
    }
  },
);
export const todolistsThunk = { editTitleTodo, addTodo, deleteTodo, setTodo };
// export const deleteTodoTC_ =
//   (id: string): ThunkActionAppType =>
//   async (dispatch) => {
//     helpChangeStatusFunc(id, "loading", dispatch);
//     try {
//       await todolistsApi.deleteTodolist(id);
//       dispatch(deleteTodoAC({ tId: id }));
//       helpChangeStatusFunc(id, "succeeded", dispatch);
//       dispatch(setInfoAC({ message: { succeededInfo: "Delete Todo!" } }));
//     } catch (error) {
//       errorCatchHelpFunc(dispatch, error);
//       dispatch(changeStatusTodoAC({ tId: id, status: "failed" }));
//     }
//   };

// export const setTodoTC_ = (): ThunkActionAppType => async (dispatch) => {
//   dispatch(changeAppStatusAC({ status: "loading" }));
//   try {
//     const res = await todolistsApi.getTodolists();
//     dispatch(setTodoAC({ todolists: res.data }));
//     dispatch(changeAppStatusAC({ status: "succeeded" }));
//   } catch (error) {
//     errorCatchHelpFunc(dispatch, error);
//   }
// };

// export const addTodoTC_ =
//   (title: string): ThunkActionAppType =>
//   async (dispatch) => {
//     dispatch(changeAppStatusAC({ status: "loading" }));
//     try {
//       const res = await todolistsApi.addTodolist(title);
//       helpResultFunc<{ item: TodolistsResponse }>(
//         () => dispatch(addTodoAC({ tId: res.data.data.item.id, title })),
//         res,
//         dispatch,
//         "Add Todo!",
//       );
//     } catch (error) {
//       errorCatchHelpFunc(dispatch, error);
//     }
//   };

// export const editTitleTodoTC_ =
//   (id: string, title: string): ThunkActionAppType =>
//   async (dispatch) => {
//     dispatch(changeAppStatusAC({ status: "loading" }));
//     try {
//       const res = await todolistsApi.changeTitleTodolist(id, title);
//       helpResultFunc(() => dispatch(editTitleTodoAC({ tId: id, title })), res, dispatch, "Todo changed!");
//     } catch (error) {
//       errorCatchHelpFunc(dispatch, error);
//     }
//   };

// export const todolistReducer = (
//   state: ChangeTodolistsResponse[] = initialState,
//   action: ActionsTodolistsType,
// ): ChangeTodolistsResponse[] => {
//   switch (action.type) {
//     case "DELETE-TODO": {
//       return state.filter((el) => el.id !== action.tId);
//     }
//     case "ADD-TODO": {
//       return [
//         {
//           id: action.tId,
//           title: action.title,
//           addedDate: "",
//           order: 0,
//           statusTodo: "idle",
//         },
//         ...state,
//       ];
//     }
//     case "EDIT-TITLE": {
//       return state.map((el) => (el.id === action.tId ? { ...el, title: action.title } : el));
//     }
//     case "SET-TODO": {
//       return action.todolists.map((el) => ({ ...el, statusTodo: "idle" }));
//     }
//     case "CHANGE-STATUS-TODO": {
//       return state.map((el) => (el.id === action.tId ? { ...el, statusTodo: action.status } : el));
//     }
//     case "DELETE-STATE": {
//       return [];
//     }
//     default:
//       return state;
//   }
// };

// export const deleteTodoAC = (tId: string) => ({ type: "DELETE-TODO", tId }) as const;
// export const addTodoAC = (tId: string, title: string) => ({ type: "ADD-TODO", tId, title }) as const;
// export const editTitleTodoAC = (tId: string, title: string) => ({ type: "EDIT-TITLE", tId, title }) as const;
// export const setTodoAC = (todolists: TodolistsResponse[]) => ({ type: "SET-TODO", todolists }) as const;
// export const changeStatusTodoAC = (tId: string, status: StatusType) =>
//   ({ type: "CHANGE-STATUS-TODO", tId, status }) as const;

// export type ActionsTodolistsType =
//   | DeleteTodoTypeAC
//   | AddTodoTypeAC
//   | EditTitleTodoTypeAC
//   | SetTodoTypeAC
//   | ChangeStatusTodoTypeAC
//   | DeleteStateTypeAC;

// export type DeleteTodoTypeAC = ReturnType<typeof deleteTodoAC>;
// export type AddTodoTypeAC = ReturnType<typeof addTodoAC>;
// export type EditTitleTodoTypeAC = ReturnType<typeof editTitleTodoAC>;
// export type SetTodoTypeAC = ReturnType<typeof setTodoAC>;
// export type ChangeStatusTodoTypeAC = ReturnType<typeof changeStatusTodoAC>;
