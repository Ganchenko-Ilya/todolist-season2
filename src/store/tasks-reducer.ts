import { v1 } from "uuid";

import { AddTodoTypeAC, DeleteTodoTypeAC, todolist1Id, todolist2Id } from "./todolists-reducer";

export type ActionsTasksType =
  | AddTaskTypeAC
  | ChangeStatusTaskTypeAC
  | DeleteTaskTypeAC
  | EditTitleTaskTypeAC
  | AddTodoTypeAC
  | DeleteTodoTypeAC;
export type TasksType = {
  id: string;
  title: string;
  isDone: boolean;
};
export type TaskObjType = {
  [key: string]: TasksType[];
};

export const initialTasksState: TaskObjType = {
  [todolist1Id]: [
    { id: v1(), title: "Css", isDone: true },
    { id: v1(), title: "React", isDone: false },
    { id: v1(), title: "TypeScript", isDone: false },
  ],
  [todolist2Id]: [
    { id: v1(), title: "book", isDone: true },
    { id: v1(), title: "bread", isDone: false },
    { id: v1(), title: "milk", isDone: false },
  ],
};
export const tasksReducer = (
  state: TaskObjType = initialTasksState,
  action: ActionsTasksType
): TaskObjType => {
  switch (action.type) {
    case "ADD-TASK": {
      return {
        ...state,
        [action.tId]: [...state[action.tId], { id: v1(), title: action.title, isDone: false }],
      };
    }
    case "CHANGE-STATUS-TASK": {
      return {
        ...state,
        [action.tId]: state[action.tId].map((el) =>
          el.id === action.id ? { ...el, isDone: action.isDone } : el
        ),
      };
    }
    case "DELETE-TASK": {
      return { ...state, [action.tId]: state[action.tId].filter((el) => el.id !== action.id) };
    }
    case "EDIT-TITLE-TASK": {
      return {
        ...state,
        [action.tId]: state[action.tId].map((el) =>
          el.id === action.id ? { ...el, title: action.title } : el
        ),
      };
    }
    case "ADD-TODO": {
      return { ...state, [action.tId]: [] };
    }
    case "DELETE-TODO": {
      const newObj = { ...state };
      delete newObj[action.tId];
      return newObj;
    }
    default:
      return state;
  }
};

export const addTaskAC = (tId: string, title: string) =>
  ({ type: "ADD-TASK", tId, title } as const);

export const changeStatusTaskAC = (tId: string, id: string, isDone: boolean) =>
  ({ type: "CHANGE-STATUS-TASK", tId, id, isDone } as const);

export const deleteTaskAC = (tId: string, id: string) =>
  ({ type: "DELETE-TASK", tId, id } as const);

export const editTitleTaskAC = (tId: string, id: string, title: string) =>
  ({ type: "EDIT-TITLE-TASK", tId, id, title } as const);

export type AddTaskTypeAC = ReturnType<typeof addTaskAC>;

export type ChangeStatusTaskTypeAC = ReturnType<typeof changeStatusTaskAC>;

export type DeleteTaskTypeAC = ReturnType<typeof deleteTaskAC>;

export type EditTitleTaskTypeAC = ReturnType<typeof editTitleTaskAC>;
