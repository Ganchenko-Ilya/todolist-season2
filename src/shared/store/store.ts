import { combineReducers } from "redux";
import { tasksReducer } from "../model/tasks-reducer";
import { todolistReducer } from "../model/todolists-reducer";
import { thunk } from "redux-thunk";
import { appReducer } from "../model/app-reducer";
import { authReducer } from "../model/auth-reducer";
import { configureStore } from "@reduxjs/toolkit";

const rootReducer = combineReducers({
  tasks: tasksReducer,
  todolists: todolistReducer,
  app: appReducer,
  auth: authReducer,
});

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().prepend(thunk),
});

//@ts-ignore
window.store = store;
