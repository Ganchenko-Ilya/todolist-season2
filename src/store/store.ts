import { applyMiddleware, combineReducers, legacy_createStore } from "redux";
import { ActionsTasksType, tasksReducer } from "./tasks-reducer";
import { ActionsTodolistsType, todolistReducer } from "./todolists-reducer";
import { composeWithDevTools } from "redux-devtools-extension";
import { thunk, ThunkAction, ThunkActionDispatch, ThunkDispatch } from "redux-thunk";
import { useDispatch } from "react-redux";

const rootReducer = combineReducers({ tasks: tasksReducer, todolists: todolistReducer });

export const store = legacy_createStore(
  rootReducer,
  {},
  composeWithDevTools(applyMiddleware(thunk))
);
type AllAction = ActionsTodolistsType | ActionsTasksType;
export const useAppDispatch = useDispatch<ThunkDispatch<RootReducerType, unknown, AllAction>>;
export type RootReducerType = ReturnType<typeof rootReducer>;
export type ThunkActionAppType = ThunkAction<any,RootReducerType,unknown,any>

//@ts-ignore
window.store = store;
