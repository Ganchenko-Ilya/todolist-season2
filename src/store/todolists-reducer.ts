import { ChangeTodolistsResponse, TodolistsResponse, todolistsApi } from "../api/todolists-api";

import { ThunkActionAppType } from "./store";

import { StatusType, changeAppStatusAC } from "./app-reducer";
import { helpChangeStatusFunc } from "./helpFunction/changeStatusHelpFunc";
import { errorCatchHelpFunc, errorHelpFunc } from "./helpFunction/errorHelpFunc";

export type ActionsTodolistsType =
  | DeleteTodoTypeAC
  | AddTodoTypeAC
  | EditTitleTodoTypeAC
  | SetTodoTypeAC
  | ChangeStatusTodoTypeAC;

export const initialTodolistsState: ChangeTodolistsResponse[] = [];

export const todolistReducer = (
  state: ChangeTodolistsResponse[] = initialTodolistsState,
  action: ActionsTodolistsType
): ChangeTodolistsResponse[] => {
  switch (action.type) {
    case "DELETE-TODO": {
      return state.filter((el) => el.id !== action.tId);
    }
    case "ADD-TODO": {
      return [
        { id: action.tId, title: action.title, addedDate: "", order: 0, statusTodo: "idle" },
        ...state,
      ];
    }
    case "EDIT-TITLE": {
      return state.map((el) => (el.id === action.tId ? { ...el, title: action.title } : el));
    }
    case "SET-TODO": {
      return action.todolists.map((el) => ({ ...el, statusTodo: "idle" }));
    }
    case "CHANGE-STATUS-TODO": {
      return state.map((el) => (el.id === action.tId ? { ...el, statusTodo: action.status } : el));
    }
    default:
      return state;
  }
};

export const deleteTodoAC = (tId: string) => ({ type: "DELETE-TODO", tId }) as const;

export const addTodoAC = (tId: string, title: string) =>
  ({ type: "ADD-TODO", tId, title }) as const;
export const editTitleTodoAC = (tId: string, title: string) =>
  ({ type: "EDIT-TITLE", tId, title }) as const;
export const setTodoAC = (todolists: TodolistsResponse[]) =>
  ({ type: "SET-TODO", todolists }) as const;
export const changeStatusTodoAC = (tId: string, status: StatusType) =>
  ({ type: "CHANGE-STATUS-TODO", tId, status }) as const;

export const setTodoTC = (): ThunkActionAppType => async (dispatch) => {
  dispatch(changeAppStatusAC("loading"));
  try {
    const res = await todolistsApi.getTodolists();
    dispatch(setTodoAC(res.data));
    dispatch(changeAppStatusAC("succeeded"));
  } catch (error) {
    errorCatchHelpFunc(dispatch, error);
  }
};
export const deleteTodoTC =
  (id: string): ThunkActionAppType =>
  async (dispatch) => {
    helpChangeStatusFunc(id, "loading", dispatch);
    try {
      await todolistsApi.deleteTodolist(id);
      dispatch(deleteTodoAC(id));
      helpChangeStatusFunc(id, "succeeded", dispatch);
    } catch (error) {
      errorCatchHelpFunc(dispatch, error);
      dispatch(changeStatusTodoAC(id, "failed"));
    }
  };
export const addTodoTC =
  (title: string): ThunkActionAppType =>
  async (dispatch) => {
    dispatch(changeAppStatusAC("loading"));
    try {
      const res = await todolistsApi.addTodolist(title);
      errorHelpFunc<{ item: TodolistsResponse }>(
        () => dispatch(addTodoAC(res.data.data.item.id, title)),
        res,
        dispatch
      );
    } catch (error) {
      errorCatchHelpFunc(dispatch, error);
    }
  };
export const editTitleTodoTC =
  (id: string, title: string): ThunkActionAppType =>
  async (dispatch) => {
    dispatch(changeAppStatusAC("loading"));
    try {
      const res = await todolistsApi.changeTitleTodolist(id, title);
      errorHelpFunc(() => dispatch(editTitleTodoAC(id, title)), res, dispatch);

      dispatch(changeAppStatusAC("succeeded"));
    } catch (error) {
      errorCatchHelpFunc(dispatch, error);
    }
  };

export type DeleteTodoTypeAC = ReturnType<typeof deleteTodoAC>;
export type AddTodoTypeAC = ReturnType<typeof addTodoAC>;
export type EditTitleTodoTypeAC = ReturnType<typeof editTitleTodoAC>;
export type SetTodoTypeAC = ReturnType<typeof setTodoAC>;
export type ChangeStatusTodoTypeAC = ReturnType<typeof changeStatusTodoAC>;
