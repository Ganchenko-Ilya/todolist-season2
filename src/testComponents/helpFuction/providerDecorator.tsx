import { Provider } from "react-redux";
import { store } from "../../store/store";
import  { ComponentType } from "react";

import App from "../../components/app/App";
import { Todolist } from "../../components/todolist/Todolist";

export const providerDecorator =
  <T extends object>(Components: ComponentType<T>):ComponentType<T> =>
  (props: T):JSX.Element => (
    <Provider store={store}>
      <Components {...props} />
    </Provider>
  );

export const AppWithProvider = providerDecorator(App);
export const TodolistWithProvider = providerDecorator(Todolist);

