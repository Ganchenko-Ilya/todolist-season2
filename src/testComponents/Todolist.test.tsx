import renderer from "react-test-renderer";
import { TodolistWithProvider } from "./helpFuction/providerDecorator";
import { initialTodolistsState } from "../store/todolists-reducer";

test("renders correctly", () => {
  const result = renderer
    .create(
      <TodolistWithProvider
        deleteTodo={() => {}}
        editTitleTodo={() => {}}
        tId={initialTodolistsState[0].id}
        titleTodo="Test"
      />
    )
    .toJSON();
  expect(result).toMatchSnapshot();
});
