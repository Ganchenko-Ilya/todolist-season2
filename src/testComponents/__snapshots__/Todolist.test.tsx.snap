// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="baseStyleTodo"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 paper css-3su884-MuiPaper-root"
  >
    <div
      className="headerTodo"
    >
      <h3>
        <div
          className="editWrapper"
        >
          <span
            onDoubleClick={[Function]}
          >
            Test
          </span>
        </div>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          onClick={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
          />
        </svg>
      </h3>
    </div>
    <div
      className="wrapper"
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        onKeyDown={[Function]}
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id=":r0:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Add Task"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-ihdtdm"
            >
              <span
                className="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
        />
      </svg>
    </div>
    <div
      className="wrapperListTasks"
    >
      <ul>
        <div
          className="wrapperTasks"
        >
          <li
            className="taskIsDone"
          >
            <div
              className="editWrapper"
            >
              <span
                onDoubleClick={[Function]}
              >
                Css
              </span>
            </div>
            <div
              className="wrapperCheckbox"
            >
              <span
                className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall css-1oxcmra-MuiButtonBase-root-MuiCheckbox-root"
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <input
                  checked={true}
                  className="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate={false}
                  onChange={[Function]}
                  required={false}
                  type="checkbox"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="CheckBoxIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
              </span>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteIcon"
                focusable="false"
                onClick={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
                />
              </svg>
            </div>
          </li>
        </div>
        <div
          className="wrapperTasks"
        >
          <li
            className=""
          >
            <div
              className="editWrapper"
            >
              <span
                onDoubleClick={[Function]}
              >
                React
              </span>
            </div>
            <div
              className="wrapperCheckbox"
            >
              <span
                className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall css-1oxcmra-MuiButtonBase-root-MuiCheckbox-root"
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <input
                  checked={false}
                  className="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate={false}
                  onChange={[Function]}
                  required={false}
                  type="checkbox"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteIcon"
                focusable="false"
                onClick={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
                />
              </svg>
            </div>
          </li>
        </div>
        <div
          className="wrapperTasks"
        >
          <li
            className=""
          >
            <div
              className="editWrapper"
            >
              <span
                onDoubleClick={[Function]}
              >
                TypeScript
              </span>
            </div>
            <div
              className="wrapperCheckbox"
            >
              <span
                className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall css-1oxcmra-MuiButtonBase-root-MuiCheckbox-root"
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <input
                  checked={false}
                  className="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate={false}
                  onChange={[Function]}
                  required={false}
                  type="checkbox"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteIcon"
                focusable="false"
                onClick={[Function]}
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
                />
              </svg>
            </div>
          </li>
        </div>
      </ul>
    </div>
    <div
      className="buttonFilters"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-11qr2p8-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        All
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-1k23hlb-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Active
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-1k23hlb-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Complited
      </button>
    </div>
  </div>
</div>
`;
